app/src/main/java/xterminators/spellingbee/model/HighScores.java

```diff
@@ -0,0 +1,148 @@
+     * Returns the current high scores.
+     */
+    public TreeMap<String, Integer> getScores() {
+        return scores;
+    }
+
+    /**
+     * Loads the current high scores from the HighScores.json file.
+     */
+    public static void loadScores() {
+        File scoreFile = new File(System.getProperty("user.home") + 
+            File.separator + FILE_NAME);
+
+        if (!scoreFile.exists()) {
+            System.out.println("score file doesn't exist");
+            return;
+        }
+
+        try {
+            Scanner reader = new Scanner(scoreFile);
+
+            StringBuilder json = new StringBuilder();
+
+            while(reader.hasNextLine()) {
+                json.append(reader.nextLine());
+            }
+
+            reader.close();
+
+            Gson gson = new Gson();
+            Type type = new TypeToken<TreeMap<String, Integer>>() {}.getType();
+            TreeMap<String, Integer> unsortedScores = 
+                gson.fromJson(json.toString(), type);
+
+            scores = 
+                new TreeMap<>(new ValueComparator(unsortedScores));
+            scores.putAll(unsortedScores);
+
+            if (scores.size() > MAX_STORED_SCORES) {
+                for (int i = MAX_STORED_SCORES; i < scores.size(); ++i) {
+                    scores.pollLastEntry();
+                }
+            }
+
+            for (Integer value : scores.values()) {
+                if (value < lowestHighScore)
+                    lowestHighScore = value;
+            }
+
+        } catch (FileNotFoundException ex) {
+            // this can be ignored because it cannot ever be thrown
+            // it could potentially be thrown by new Scanner()
+        }
+
+        if (lowestHighScore == Integer.MAX_VALUE) {
+            // We could get here if there are no high scores saved
+            // We are assuming that the user is not capable
+            // of getting Integer.MAX_VALUE points
+            lowestHighScore = 0;
+        }
+    }
+
+    /**
+     * Saves the current scores to the HighScores.json file.
+     */
+    public boolean saveScores() {
+        try {
+            String userHome = System.getProperty("user.home");
+            String filePath = userHome + File.separator + FILE_NAME;
+            File saveLocation = new File(filePath);
+
+            Gson gson = new Gson();
+            String json = gson.toJson(scores);
+
+            FileWriter writer = new FileWriter(saveLocation);
+            writer.write(json);
+            writer.close();
+        } catch (Exception ex) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Adds a high score with the given username and score
+     * to the high scores list, and then saves the high scores
+     * list to the high scores file.
+     * 
+     * @param userName The username to save the high score with.
+     * @param score The score of the high score.
+     */
+    public boolean saveScore(String userName, int score) {
```

Comment:
Do we not want to also save the name of the puzzle associated with that score? or the word used to form the puzzle.